src/app/file/svg_format.cpp

```diff
@@ -0,0 +1,161 @@
+
+  int onGetFlags() const override {
+    return
+      FILE_SUPPORT_SAVE |
+      FILE_SUPPORT_RGB |
+      FILE_SUPPORT_RGBA |
+      FILE_SUPPORT_GRAY |
+      FILE_SUPPORT_INDEXED |
+      FILE_SUPPORT_SEQUENCES;
+  }
+
+  bool onLoad(FileOp* fop) override;
+#ifdef ENABLE_SAVE
+  bool onSave(FileOp* fop) override;
+#endif
+};
+
+FileFormat* CreateSvgFormat()
+{
+  return new SvgFormat;
+}
+
+  bool SvgFormat::onLoad(FileOp* fop) { return false;}
+
+#ifdef ENABLE_SAVE
+
+bool SvgFormat::onSave(FileOp* fop)
+{
+  const Image* image = fop->sequenceImage();
+  unsigned char image_palette[256][4];
+  int x, y, c, r, g, b, a, alpha;
+  float opacity;
+  FileHandle handle(open_file_with_exception_sync_on_close(fop->filename(), "wb"));
+  FILE* f = handle.get();
+
+  if (image->pixelFormat() == IMAGE_INDEXED) {
+    for (y=0;y<256; y++) {
+      fop->sequenceGetColor(y, &r, &g, &b);
+      image_palette[y][0] = r;
+      image_palette[y][1] = g;
+      image_palette[y][2] = b;
+      fop->sequenceGetAlpha(y, &a);
+      image_palette[y][3] = a;
+    }
+  }
+  fprintf(f, "<?xml version=\"1.0\" encoding=\"UTF-8\" ?>\n");
+  fprintf(f, "<svg version=\"1.1\" width=\"%d\" height=\"%d\" xmlns=\"http://www.w3.org/2000/svg\">\n",
+          image->width(), image->height());
+  switch (image->pixelFormat()) {
+
+    case IMAGE_RGB:
+      for (y=0; y<image->height(); y++) {
+        for (x=0; x<image->width(); x++) {
+          c = get_pixel_fast<RgbTraits>(image, x, y);
+          alpha = rgba_geta(c);
+          if (alpha != 0x00) {
+            fprintf(f, "<rect x=\"%d\" y=\"%d\" width=\"1\" height=\"1\" fill=\"#", x, y);
+            fprintf(f, "%02X%02X%02X\" ", rgba_getr(c), rgba_getg(c), rgba_getb(c));
+            if (alpha != 0xff) {
+              opacity = (float)alpha / 255.0;
+              fprintf(f, "opacity=\"%f\" ", opacity);
+            }
+            fprintf(f, "/>\n");
+          }
+        }
+        fop->setProgress((float)y / (float)(image->height()));
+      }
+      break;
+
+    case IMAGE_GRAYSCALE:
+      for (y=0; y<image->height(); y++) {
+        for (x=0; x<image->width(); x++) {
+          c = get_pixel_fast<GrayscaleTraits>(image, x, y);
+          auto v = graya_getv(c);
+          alpha = graya_geta(c);
+          if (alpha != 0x00) {
+            fprintf(f, "<rect x=\"%d\" y=\"%d\" width=\"1\" height=\"1\" fill=\"#", x, y);
+            fprintf(f, "%02X%02X%02X\" ", v, v, v);
+            if (alpha != 0xff) {
+              opacity = (float)alpha / 255.0;
+              fprintf(f, "opacity=\"%f\" ", opacity);
+            }
+            fprintf(f, "/>\n");
+          }
+        }
+        fop->setProgress((float)y / (float)(image->height()));
+      }
+      break;
+
+    case IMAGE_INDEXED:
+      if (fop->document()->sprite()->backgroundLayer() != NULL &&
+          fop->document()->sprite()->backgroundLayer()->isVisible()) {
+        fprintf(f, "<rect x=\"0\" y=\"0\" width=\"%d\" height=\"%d\" fill=\"#000000\" />\n",
```

Comment:
The background color is not always black (`#000000`). You have to use the palette color.